src/palace.pyx

```diff
@@ -367,13 +371,16 @@ cdef class Context:
             shared_ptr[alure.MessageHandler](new CppMessageHandler(self)))
 
     def __enter__(self) -> Context:
+        Context.__context_stack.append(self)
```

Comment:
Not saying I agree with this approach, but if you use it, keep it in `use_context`.

That being said, it's not our responsibility (or capability) to keep track of current context.  E.g. what if `use_context` have problems running?  When possible, fetch info from alure's C++ API, e.g. `self.previous = self.impl.get_current()`.

Reply:
But can we get `device` from `get_current` though? Since it's a constructor argument I assume it's part of the identity of a context.

Reply:
Device can be retrieved from `Context::getDevice`, but you have a point tho: the message handler instance of the context would be lost.  But no worries about it now, just open an issue describing it and we can solve it later.

Wait this is not about the function `current_context` sorry.  All you have to do is to put the exact statement I gave above here.  Now, about `current_context`, you'll need to allow the Python `Context` to be created without any initialization (see `None` idioms in other classes), then assign possible fields in there.  Message handler can be handled later.

Reply:
A context initialized without a device would raise an error because it's not optional in the constructor, so I guess I have to do it.